src/Common/Dust.cpp

```diff
@@ -660,128 +636,129 @@ void DustSphNgbFinder<ndim, ParticleType>::FindNeibAndDoForces
 
-      // Loop over all active particles in the cell
-      //-------------------------------------------------------------------------------------------
-      for (j=0; j<Nactive; j++) {
-        i = activelist[j];
+        // Loop over all active particles in the cell
+        //-------------------------------------------------------------------------------------------
+        for (j=0; j<Nactive; j++) {
+          i = activelist[j];
 
-        if (types[activepart[j].ptype].drag_forces) {
+          if (types[activepart[j].ptype].drag_forces) {
 
-          Typemask dragmask = types[activepart[j].ptype].dragmask;
+            Typemask dragmask = types[activepart[j].ptype].dragmask;
 
-          const bool do_pair_once=false;
-          NeighbourList<ParticleType<ndim> > neiblist =
-              neibmanager.GetParticleNeib(activepart[j],dragmask,do_pair_once);
+            const bool do_pair_once=false;
+            NeighbourList<ParticleType<ndim> > neiblist =
+                neibmanager.GetParticleNeib(activepart[j],dragmask,do_pair_once);
 
 #if defined(VERIFY_ALL)
-          neibmanager.VerifyNeighbourList(i, hydro->Nhydro, sphdata, "all");
-          neibmanager.VerifyReducedNeighbourList(i, neiblist, hydro->Nhydro, sphdata, dragmask, "all");
+            neibmanager.VerifyNeighbourList(i, hydro->Nhydro, sphdata, "all");
+            neibmanager.VerifyReducedNeighbourList(i, neiblist, hydro->Nhydro, sphdata, dragmask, "all");
 #endif
 
+            DOUBLE dt_drag = drag_timestep(activepart[j]);
 
-          DOUBLE dt_drag = drag_timestep(activepart[j]);
+            Forces.ComputeDragForces(activepart[j],neiblist, dt_drag, &(a_drag[i*ndim]));
+          }
 
-          Forces.ComputeDragForces(activepart[j],neiblist, dt_drag, &(a_drag[i*ndim]));
         }
 
+        // Add all active particles contributions to main array
+        for (j=0; j<Nactive; j++) {
+          i = activelist[j];
+          sphdata[i].div_v = activepart[j].div_v ;
+          sphdata[i].sound = activepart[j].sound ;
+          dudt_local[i] += activepart[j].dudt ;
+        }
+        for (j=0; j<Nneib; j++) {
+          std::pair<int,ParticleType<ndim>*> neighbour=neibmanager.GetNeibI(j);
+          dudt_local[neighbour.first] += neighbour.second->dudt ;
+        }
       }
 
-      // Add all active particles contributions to main array
-      for (j=0; j<Nactive; j++) {
-        i = activelist[j];
-        sphdata[i].div_v = activepart[j].div_v ;
-        sphdata[i].sound = activepart[j].sound ;
-        dudt_local[i] += activepart[j].dudt ;
-      }
-      for (j=0; j<Nneib; j++) {
-        std::pair<int,ParticleType<ndim>*> neighbour=neibmanager.GetNeibI(j);
-        dudt_local[neighbour.first] += neighbour.second->dudt ;
-      }
-    }
-
 #pragma omp critical
-    {
-      for (i=0; i<hydro->Ntot; i++) {
-        dudt[i] += dudt_local[i] ;
+      {
+        for (i=0; i<hydro->Ntot; i++) {
+          dudt[i] += dudt_local[i] ;
```

Comment:
I've done this (needs checking though).  I think a barrier is still needed after the MPI communication.